docs/running-on-mesos.md

```diff
@@ -211,6 +211,16 @@ See the [configuration page](configuration.html) for information on Spark config
   </td>
 </tr>
 <tr>
+  <td><code>spark.mesos.executor.cores</code></td>
```

Comment:
Similarly, the naming of this config is misleading. In the rest of Spark we use the dot notation to denote hierarchy. For instance, we have `spark.yarn.am.waitTime`, which falls under the `spark.yarn.*` namespace. In this case, people might interpret this as the number of cores to give the Spark executor, since this config falls under the `spark.mesos.*` namespace and elsewhere "executor" means Spark executors.

So we should probably rename this to `spark.mesos.mesosExecutor.cores`. Unfortunately, this needs to start with `spark.mesos.*` to keep it consistent with other Mesos-specific configs, and it needs to be `mesosExecutor` because otherwise people may interpret it as Spark executor cores instead. It's a little more verbose but much less ambiguous.